src/liballoc/rc.rs

```diff
@@ -782,8 +782,9 @@ impl<T: ?Sized> Drop for Weak<T> {
     fn drop(&mut self) {
         unsafe {
             let ptr = *self._ptr;
-            if !(*(&ptr as *const _ as *const *const ())).is_null() &&
-               ptr as *const () as usize != mem::POST_DROP_USIZE {
+            let thin = ptr as *const ();
+
+            if !(thin as *const *const ()).is_null() && thin as usize != mem::POST_DROP_USIZE {
```

Comment:
I had re-added it when @aturon pointed out that I was missing the second level. I thought it was pointless, but I thought it might have been causing the segfaults.


On Dec 29, 2015, 19:30 -0500, Alexis Beingessner<notifications@github.com>, wrote:
>  
> Insrc/liballoc/rc.rs(https://github.com/rust-lang/rust/pull/30593#discussion_r48580830):
>  
> >@@ -782,8 +782,9 @@ impl<T: ?Sized>Drop for Weak<T>{>fn drop(&mut self) {>unsafe {>let ptr = *self._ptr;>-      if !(*(&ptr as *const _ as *const *const ())).is_null()&&>-        ptr as *const () as usize != mem::POST_DROP_USIZE {>+      let thin = ptr as *const ();>+>+      if !(thin as *const *const ()).is_null()&&thin as usize != mem::POST_DROP_USIZE {
>  
> In particular this cast seems pointless?
>  
>  
> —
> Reply to this email directly orview it on GitHub(https://github.com/rust-lang/rust/pull/30593/files#r48580830).
>  
>  
>